### PR TITLE
feat(client): add verifiers

### PR DESCRIPTION
**Pull request by javadkh2**
Adding support for verifiers based on [verifiers proposal](https://github.com/kadena-io/KIPs/pull/57/files)

### CODE REVIEW

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts [hunk 1]
```diff
@@ -49,6 +51,28 @@ interface IAddSigner<TCommand> {
   ): IBuilder<TCommand>;
 }
 
+interface IAddVerifier<TCommand> {
+  /**
+   * Add signer without capability
```
Comment (edmundnoble) [inline packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts:56]: s/signer/verifier

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts [hunk 2]
```diff
@@ -49,6 +51,28 @@ interface IAddSigner<TCommand> {
...
+   */
+  (first: IVerifier): IBuilder<TCommand>;
+  /**
+   * Add a signer including capabilities. The withCapability function is obtained from
```
Comment (edmundnoble) [inline packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts:60]: s/signer/verifier

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts [hunk 3]
```diff
@@ -81,6 +105,23 @@ interface IAddKeyset<TCommand> {
  * @public
  */
 export interface IBuilder<TCommand> {
+  /**
+   * Add signer with theirs capabilities
```
Comment (edmundnoble) [inline packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts:109]: s/signer/verifier, s/theirs/their

**Review by edmundnoble — APPROVED** [review]

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts (hunk 1 above)
Comment (javadkh2) [reply packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts]: fixed

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts (hunk 2 above)
Comment (javadkh2) [reply packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts:60]: fixed

File: packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts (hunk 3 above)
Comment (javadkh2) [reply packages/libs/client/src/createTransactionBuilder/createTransactionBuilder.ts]: fixed

File: packages/libs/client/README.md [hunk 4]
```diff
@@ -245,9 +247,41 @@ const signedTransaction = signWithChainweaver(unsignedTransaction)
 There's several steps to setup a WalletConnect connections and sign with
 WalletConnect.
 
-1. Setting up the connection using [`ClientContextProvider.tsx`][30]
+1. Setting up the connection using [`ClientContextProvider.tsx`][32]
 2. Use `signWithWalletConnect` to request a signature from the wallet
-   (`Transaction.tsx`)\[[https://github.com/kadena-io/wallet-connect-example/blob/2efc34296f845aea75f37ab401a5c49081f75b47/src/components/Transaction.tsx#L104][31]]
+   (`Transaction.tsx`)\[[https://github.com/kadena-io/wallet-connect-example/blob/2efc34296f845aea75f37ab401a5c49081f75b47/src/components/Transaction.tsx#L104][33]]
+
+## Verifier
```
Comment (javadkh2) [inline packages/libs/client/README.md:254]: @edmundnoble 
Is this explanation correct?

File: packages/libs/client/README.md [hunk 5]
```diff
@@ -245,9 +247,41 @@ const signedTransaction = signWithChainweaver(unsignedTransaction)
...
+
+Kadena supports `verifier` as another way of gaining authority in Pact, as well
+as the normal signature flow. In this way, a `verifier`, `proof`, and the
+`capability list` will be sent to the blockchain, and if the proof satisfies the
+verifier function, the capabilities will be granted. It's useful, for example,
+for ZK (Zero Knowledge Proof) or bridging between networks.
```
Comment (javadkh2) [inline packages/libs/client/README.md:260]: @edmundnoble 
Is this explanation right?

**Comment by Randynamic** [comment]
❤️

**Review by Randynamic — APPROVED** [review]

File: packages/libs/client/README.md (hunk 5 above)
Comment (edmundnoble) [reply packages/libs/client/README.md:260]: This is good yes.

**Review by alber70g — APPROVED** [review]

**Pull request merged**